read-fonts/src/table_ref.rs

```diff
@@ -6,6 +6,7 @@ use crate::{
     offset::{Offset, ResolveOffset},
 };
 
+#[derive(Clone)]
```

Comment:
I had originally avoided adding this intentionally to enforce some abstract notion of purity, but... it's probably fine? The only thing we're going to be cloning is the marker, but we might want to discourage having this be used widely?

If so, and if we need this, I previously had a private method called `sneaky_copy` that i used in the traversal code.

Reply:
I’m not sure. As above, I like having `Clone` available if it doesn’t break any invariants. I would personally make all of these types `Copy` as well but I found this to be a decent middle ground :)